crates/bevy_render/src/lib.rs

```diff
@@ -102,13 +108,32 @@ use crate::{
 };
 use alloc::sync::Arc;
 use bevy_app::{App, AppLabel, Plugin, SubApp};
-use bevy_asset::{load_internal_asset, weak_handle, AssetApp, AssetServer, Handle};
+use bevy_asset::{AssetApp, AssetServer};
 use bevy_ecs::{prelude::*, schedule::ScheduleLabel};
 use bitflags::bitflags;
 use core::ops::{Deref, DerefMut};
 use std::sync::Mutex;
 use tracing::debug;
 
+// TODO: this should be replaced by proper asset handling in the shader loader.
```

Comment:
I'm a little skeptical this will ever be addressed. In fact it sounds like we're gonna be leaning in harder by making shaders use less of the asset system? Up to you whether to keep this TODO.

Some discussion about that: https://discord.com/channels/691052431525675048/749332104487108618/1369108776472416286

Reply:
yote